File: apps/etterlatte-saksbehandling-ui/client/src/shared/types/sanksjon.ts
export interface ISanksjon {
  id?: string
  behandlingId: string
  sakId: number
  type: SanksjonType
  fom: string
  tom?: string
  opprettet: {
    tidspunkt: string
    ident: string
  }
  endret?: {
    tidspunkt: string
    ident: string
  }
  beskrivelse: string
}

export interface ISanksjonLagre {
  id?: string
  sakId: number
  type: SanksjonType
  fom: string
  tom?: string
  beskrivelse: string
}

export enum SanksjonType {
  BORTFALL = 'BORTFALL',
  OPPHOER = 'OPPHOER',
  STANS = 'STANS',
  UTESTENGING = 'UTESTENGING',
  IKKE_INNVILGET_PERIODE = 'IKKE_INNVILGET_PERIODE',
}

export const valgbareSanksjonstyper = [
  SanksjonType.BORTFALL,
  SanksjonType.OPPHOER,
  SanksjonType.STANS,
  SanksjonType.UTESTENGING,
  SanksjonType.IKKE_INNVILGET_PERIODE,
]

export const visbareSanksjonstyper = [
  SanksjonType.BORTFALL,
  SanksjonType.OPPHOER,
  SanksjonType.STANS,
  SanksjonType.UTESTENGING,
]

export const tekstSanksjon: Record<SanksjonType, string> = {
  BORTFALL: 'Bortfall',
  OPPHOER: 'Opphør',
  STANS: 'Stans',
  UTESTENGING: 'Utestenging',
  IKKE_INNVILGET_PERIODE: 'Vilkår ikke oppfylt',
}

File: libs/etterlatte-beregning-model/src/main/kotlin/Sanksjon.kt
package no.nav.etterlatte.libs.common.beregning

import no.nav.etterlatte.libs.common.grunnlag.Grunnlagsopplysning
import no.nav.etterlatte.libs.common.sak.SakId
import java.time.YearMonth
import java.util.UUID

data class SanksjonertYtelse(
    val sanksjonId: UUID,
    val sanksjonType: SanksjonType,
)

data class Sanksjon(
    val id: UUID?,
    val behandlingId: UUID,
    val sakId: SakId,
    val type: SanksjonType,
    val fom: YearMonth,
    val tom: YearMonth?,
    val opprettet: Grunnlagsopplysning.Saksbehandler,
    val endret: Grunnlagsopplysning.Saksbehandler?,
    val beskrivelse: String,
)

enum class SanksjonType {
    BORTFALL,
    OPPHOER,
    STANS,
    UTESTENGING,
    IKKE_INNVILGET_PERIODE,
}
